src/storage/src/storage_state.rs

```diff
@@ -550,7 +739,110 @@ impl<'w, A: Allocate> Worker<'w, A> {
+                    // reconciliation later.
+                    self.ingestions
+                        .insert(ingestion.id, ingestion.description.clone());
+
+                    // Initialize shared frontier reporting.
+                    for (export_id, _export) in ingestion.description.source_exports.iter() {
+                        self.reported_frontiers.insert(
+                            *export_id,
+                            Antichain::from_elem(mz_repr::Timestamp::minimum()),
+                        );
+                    }
+
+                    // This needs to be done by one worker, which will
+                    // broadcasts a `CreateIngestionDataflow` command to all
+                    // workers based on the response that contains the
+                    // resumption upper.
+                    //
+                    // Doing this separately on each worker could lead to
+                    // differing resume_uppers which might lead to all kinds of
+                    // mayhem.
+                    if worker_index == 0 {
+                        async_worker.calculate_resume_upper(ingestion.id, ingestion.description);
+                    }
+                }
+            }
+            StorageCommand::CreateSinks(exports) => {
+                for export in exports {
+                    // Remember the sink description to facilitate possible
+                    // reconciliation later.
+                    self.exports.insert(export.id, export.description.clone());
+
+                    self.reported_frontiers.insert(
+                        export.id,
+                        Antichain::from_elem(mz_repr::Timestamp::minimum()),
+                    );
+
+                    self.sink_handles.insert(
+                        export.id,
+                        SinkHandle::new(
+                            export.id,
+                            export
+                                .description
+                                .from_storage_metadata
+                                .persist_location
+                                .clone(),
+                            export.description.from_storage_metadata.data_shard,
+                            Arc::clone(&self.persist_clients),
+                        ),
+                    );
+
+                    // This needs to be broadcast by one worker and go through
+                    // the internal command fabric, to ensure consistent
+                    // ordering of dataflow rendering across all workers.
+                    if worker_index == 0 {
+                        internal_cmd_tx.broadcast(InternalStorageCommand::CreateSinkDataflow(
+                            export.id,
+                            export.description,
+                        ));
+                    }
+                }
+            }
+            StorageCommand::AllowCompaction(list) => {
```

Comment:
I think these have to be processed in order, otherwise a `SuspendAndRestart` could be interleaved with an `AllowCompaction` and leave some worker running an ingestion that should be dead

Reply:
Good catch! They also need the split-into-internal-message treatment. Though we do have to yank the `reported_frontier` here, see this comment: https://github.com/MaterializeInc/materialize/pull/16958/files?diff=split&w=0#r1072847940

Reply:
glad its actually a problem I didnt catch a false negative!